thentos-adhocracy/tests/Thentos/Adhocracy3/Backend/Api/SimpleSpec.hs

```diff
@@ -91,6 +95,40 @@ spec =
                      (Aeson.eitherDecode userdata :: Either String A3UserWithPass)
                      `shouldBe` Left "Illegal whitespace sequence in user name: \" Anna  Toll\""
 
+        describe "login" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad credentials mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "name"     .= String "Anna Müller"
+                            , "password" .= String "this-is-wrong"
+                            ]
+
+                    rsp <- srequest $ makeSRequest "POST" "login_username" [] reqBody
+                    shouldBeErr400WithCustomMessage (simpleStatus rsp) (simpleBody rsp)
+                        "\"User doesn't exist or password is wrong\""
+
+        describe "activate_account" . before setupTestBackend . after teardownTestBackend $
+            it "rejects bad path mimicking A3" $
+                \bts -> runTestBackend bts $ do
+
+                    let reqBody = encodePretty . object $
+                            [ "path" .= String "/activate/no-such-path" ]
+
+                    rsp <- srequest $ makeSRequest "POST" "activate_account" [] reqBody
```

Comment:
I don't entirely understand why we've been using `srequest` and `makeSRequest`. In this particular case, wouldn't something like

```
context "when the path doesn't exist" do
  it "responds with 400" $ do 
     post "activate_account" reqBody `shouldRespondWith` 400
  it "sets the content-type to application/json" $ do 
     post "activate_account" reqBody `shouldRespondWith` ( "Content-Type" ...)
  it "responds with the appropriate body" $ do
    post "activate_account" reqBody `shouldRespondWith` [json|...]
```
Test everything that this does (plus a little more about the actual json structure), be a little cleaner (in the sense that we'd have smaller orthogonal tests, and a more self-evident specification of the behaviour we expect), and also be shorter?